src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2952,6 +2953,174 @@ theorem sub_le_sub_iff_le {x y z : BitVec w} (hxz : z ≤ x) (hyz : z ≤ y) :
     BitVec.toNat_sub_of_le (by rw [BitVec.le_def]; omega)]
   omega
 
+/-! ### neg-/
+
+theorem msb_eq_toInt {x : BitVec w}:
+    x.msb = decide ((x.toInt) < 0) := by
+  by_cases h : x.msb <;>
+  · simp [h, toInt_eq_msb_cond]
+    omega
+
+theorem msb_eq_toNat {x : BitVec w}:
+    x.msb = decide ((x.toNat) ≥ 2 ^ (w - 1)) := by
+  simp only [msb_eq_decide, ge_iff_le]
+
+/-- carry i x y c returns true if the `i` carry bit is true when computing `x + y + c`. -/
+def carry (i : Nat) (x y : BitVec w) (c : Bool) : Bool :=
+  decide (x.toNat % 2^i + y.toNat % 2^i + c.toNat ≥ 2^i)
+
+-- Taken from BitVec/Bitblast.lean
+theorem getLsbD_add {i : Nat} (i_lt : i < w) (x y : BitVec w) :
+    getLsbD (x + y) i =
+      (getLsbD x i ^^ (getLsbD y i ^^ carry i x y false)) := by
+  sorry
+
+-- Taken from BitVec/Bitblast.lean
+theorem carry_incr (i : Nat) (x : BitVec w) (h : 0 < w) :
+    carry (i+1) x (1#w) false = decide (∀ j ≤ i, x.getLsbD j = true) := by
+  sorry
+
```

Comment:
These are existing results in the Bitblast file, which I needed to prove msb_neg